src/wallet/wallet.cpp

```diff
@@ -1868,8 +1868,18 @@ bool CWalletTx::IsTrusted() const
         if (parent == NULL)
             return false;
         const CTxOut& parentOut = parent->tx->vout[txin.prevout.n];
-        if (pwallet->IsMine(parentOut) != ISMINE_SPENDABLE)
-            return false;
+        const auto& isMine = pwallet->IsMine(parentOut);
+        if (isMine != ISMINE_SPENDABLE)
+        {
+            // If the wallet is hdwatchonly, check if it is a key we generated
+            if (!pwallet->IsHDWatchOnly() || isMine != ISMINE_WATCH_SOLVABLE)
+                return false;
+            const auto& meta = pwallet->mapKeyMetadata;
+            auto it = meta.find(CScriptID(parentOut.scriptPubKey));
+            if (it == meta.end() ||
+                it->second.hdKeypath.empty())
```

Comment:
Does `IsTrusted()`  returns `ISMINE_SPENDABLE` when for a multi sig output where we know all the keys?